pkg/kn/commands/service/import_test.go

```diff
@@ -0,0 +1,277 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"gotest.tools/assert"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	knclient "knative.dev/client/pkg/serving/v1"
+	"knative.dev/client/pkg/util"
+	"knative.dev/client/pkg/util/mock"
+	"knative.dev/client/pkg/wait"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+)
+
+func TestServiceImportFilenameError(t *testing.T) {
+	client := knclient.NewMockKnServiceClient(t)
+	r := client.Recorder()
+
+	_, err := executeServiceCommand(client, "import")
+
+	assert.Error(t, err, "'kn service import' requires filename of import file as single argument")
+	r.Validate()
+}
+
+func TestServiceImportExistError(t *testing.T) {
+	file, err := generateFile([]byte(exportYAML))
+	assert.NilError(t, err)
+	defer os.RemoveAll(filepath.Dir(file))
+
+	client := knclient.NewMockKnServiceClient(t)
+	r := client.Recorder()
+
+	r.GetService("foo", nil, nil)
+	_, err = executeServiceCommand(client, "import", file)
+
+	assert.ErrorContains(t, err, "because the service already exists")
```

Comment:
In general, for tests, I would assert that all required context information is included, i.e. here the service name and maybe the namespace (if this is part of the error message). The exact wording is secondary.

Reply:
This is true for the other checks below, too, that check for some messages